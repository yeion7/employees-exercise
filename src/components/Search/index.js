import React from 'react'
import PropTypes from 'prop-types'

const Search = ({ onSearch }) => {
  return (
    <div>
      <input
        data-test="input"
        type="search"
        placeholder="Buscar empleado o empresa"
        onChange={onSearch}
      />
    </div>
  )
}

Search.propTypes = {
  onSearch: PropTypes.func.isRequired
}

export default Search
