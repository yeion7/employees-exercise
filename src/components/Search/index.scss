.search {
  &__input {
    width: 100%;
    padding: 10px;
    margin-bottom: 10px;
    font-size: 1.2em;
  }
}
