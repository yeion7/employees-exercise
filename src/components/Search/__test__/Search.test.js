import React from 'react'
import renderer from 'react-test-renderer'

import Search from '../'

describe('Search component', () => {
  it('renders correctly', () => {
    const fn = jest.fn()

    const tree = renderer.create(<Search onSearch={fn} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
