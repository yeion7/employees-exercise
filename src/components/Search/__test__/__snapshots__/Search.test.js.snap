// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly 1`] = `
<div>
  <input
    data-test="input"
    onChange={[Function]}
    placeholder="Buscar empleado o empresa"
    type="search"
  />
</div>
`;
